src/main/java/com/sequenceiq/cloudbreak/converter/ClusterConverter.java

```diff
@@ -39,6 +64,28 @@ public Cluster convert(ClusterRequest clusterRequest) {
         return cluster;
     }
 
+    private void validateBlueprintRequest(JsonNode root, Stack stack) {
+        Iterator<JsonNode> hostGroups = root.get("host_groups").elements();
+        int hostGroupCount = 0;
+        while (hostGroups.hasNext()) {
+            JsonNode next = hostGroups.next();
+            String name = next.get("name").asText();
+            boolean find = false;
+            for (InstanceGroup instanceGroup : stack.getInstanceGroups()) {
+                if (instanceGroup.getGroupName().equals(name)) {
+                    find = true;
+                }
+            }
+            hostGroupCount++;
+            if (!find) {
+                throw new BadRequestException(String.format("Hostgroup with '%s' name was not defined as instancegroup", name));
+            }
+        }
+        if (stack.getInstanceGroups().size() != hostGroupCount) {
+            throw new BadRequestException(String.format("Request not defined all instance group on '%s' stack.", stack.getId()));
```

Comment:
Please change the error messages to:
The name of host group '%s' doesn't match any of the defined instance groups.
Every instance group must have a matching host group name in the blueprint.